src/server/pachyderm_test.go

```diff
@@ -104,6 +105,56 @@ func createTestCommits(t *testing.T, repoName, branchName string, numCommits int
 	}
 }
 
+func deleteEtcd(t *testing.T, ctx context.Context, namespace string) {
+	kubeClient := tu.GetKubeClient(t)
+	label := "app=etcd"
+	etcd, err := kubeClient.CoreV1().Pods(namespace).
+		List(ctx, metav1.ListOptions{LabelSelector: label})
+	require.NoError(t, err, "Error attempting to find etcd pod.")
+	require.Equal(t, 1, len(etcd.Items), "etcd did not have the correct number of pods.")
+
+	watcher, err := kubeClient.CoreV1().Pods(namespace).
+		Watch(ctx, metav1.ListOptions{LabelSelector: label})
+	defer watcher.Stop()
+	require.NoError(t, err)
```

Comment:
After you add t.Helper, you'll want to make sure that every failure is annotated.  (`"starting pod watch should succeed"` etc.)